doozer/doozerlib/cli/scan_sources.py

```diff
@@ -628,18 +628,51 @@ def _detect_rhcos_status(self) -> list:
                     latest_rhcos_value = self._latest_rhcos_build_id(version, arch, private)
 
-                if not latest_rhcos_value:
-                    status['changed'] = False
-                    status['reason'] = "could not find an RHCOS build to sync"
-                elif tagged_rhcos_value == latest_rhcos_value:
-                    status['changed'] = False
-                    status['reason'] = f"latest RHCOS build is still {latest_rhcos_value} -- no change from istag"
-                else:
+                if latest_rhcos_value and tagged_rhcos_value != latest_rhcos_value:
+                    status['updated'] = True
                     status['changed'] = True
                     status['reason'] = (
                         f"latest RHCOS build is {latest_rhcos_value} which differs from istag {tagged_rhcos_value}"
                     )
-                if status['changed']:
+                    statuses.append(status)
+                # check outdate rpms in rhcos
+                pullspec_for_tag = dict()
+                build_id = ""
+                for container_conf in self.runtime.group_config.rhcos.payload_tags:
+                    build_id, pullspec = rhcos.RHCOSBuildFinder(self.runtime, version, arch, private).latest_container(
+                        container_conf
+                    )
+                    pullspec_for_tag[container_conf.name] = pullspec
+                non_latest_rpms = await rhcos.RHCOSBuildInspector(
+                    self.runtime, pullspec_for_tag, arch, build_id
+                ).find_non_latest_rpms()
+                non_latest_rpms_filtered = []
+                if self.runtime.group_config.rhcos.get("layered_rhcos", False):
+                    # exclude rpm if non_latest_rpms in rhel image rpm list
+                    exclude_rpms = self.runtime.group_config.rhcos.get("exempt_rpms", [])
+                    for installed_rpm, latest_rpm, repo in non_latest_rpms:
+                        if any(excluded in installed_rpm for excluded in exclude_rpms):
+                            self.runtime.logger.info(
+                                f"[EXEMPT SKIPPED] Exclude {installed_rpm} because its in the exempt list when {latest_rpm} was available in repo {repo}"
+                            )
+                        else:
+                            non_latest_rpms_filtered.append((installed_rpm, latest_rpm, repo))
+                else:
+                    non_latest_rpms_filtered = non_latest_rpms
+                if non_latest_rpms_filtered:
+                    status['outdated'] = True
+                    status['changed'] = True
+                    status['rhelupdate'] = any(
+                        any(
+                            repo_name in repo
+                            for repo_name in ['rhel-96-baseos', 'rhel-96-appstream', 'rhel-9-fast-datapath-rpms']
```

Comment:
CXamn we get rid of these hardcoded values?

Reply:
These are rhel image build repos https://github.com/coreos/rhel-coreos-config/blob/main/manifest-rhel-9.6.yaml#L16-L19, I don't think it changes often